jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -42,7 +42,8 @@
  * The length should be representable with int32_t.
  */
 JERRY_STATIC_ASSERT ((uint32_t) ((int32_t) ECMA_STRING_MAX_CONCATENATION_LENGTH) ==
-                     ECMA_STRING_MAX_CONCATENATION_LENGTH);
+                     ECMA_STRING_MAX_CONCATENATION_LENGTH,
+                     the_length_should_be_representable_with_int32_t);
```

Comment:
`ECMA_STRING_MAX_CONCATENATION_LENGTH_should_be_`...
(And why that double type cast? It surely must have a tricky reason but it's not obvious to me.)